tests/devices/test_default_mixed.py

```diff
@@ -987,6 +987,110 @@ def test_apply_pauli_error(self, tol):
 
         assert np.allclose(dev.state, target, atol=tol, rtol=0)
 
+class TestReadoutError:
+    """Tests for measurement readout error"""
+
+    @pytest.mark.parametrize("prob", [0,0.5,1])
+    def test_state(self,prob):
+        """Tests the state output is not affected by readout error"""
+        dev = qml.device("default.mixed",wires=1,readout_prob=prob)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.state()
+
+        res=circuit()
+        expected = np.array([[1.+0.j, 0.+0.j],[0.+0.j, 0.+0.j]])
+        assert np.allclose(res,expected)
+
+    @pytest.mark.parametrize("prob", [0,0.5,1])
+    def test_density_matrix(self,prob):
+        """Tests the density matrix output is not affected by readout error"""
+        dev = qml.device("default.mixed",wires=2,readout_prob=prob)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.density_matrix(wires=0)
+
+        res=circuit()
+        expected = np.array([[1.+0.j, 0.+0.j],[0.+0.j, 0.+0.j]])
+        assert np.allclose(res,expected)
+
+    @pytest.mark.parametrize("nr_wires", [2,3])
+    def test_readout_prob_0(self,nr_wires):
+        """Tests the measurement results for readout error probability=0"""
+        dev = qml.device("default.mixed",wires=nr_wires,readout_prob=0)
+
+        @qml.qnode(dev)
+        def circuit1():
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        res=circuit1()
+        expected = np.array([1, 1])
+        assert np.allclose(res,expected)
+
+        @qml.qnode(dev)
+        def circuit2():
+            return qml.probs(wires=0)
+
+        res=circuit2()
+        expected = np.array([1, 0])
+        assert np.allclose(res, expected)
+
+    @pytest.mark.parametrize("nr_wires", [2,3])
+    def test_readout_prob_1(self,nr_wires):
+        """Tests the measurement results when readout error probability=1"""
+        dev = qml.device("default.mixed",wires=nr_wires,readout_prob=1)
+
+        @qml.qnode(dev)
+        def circuit1():
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        res=circuit1()
+        expected = np.array([-1,-1])
+        assert np.allclose(res,expected)
+
+        @qml.qnode(dev)
+        def circuit2():
+            return qml.probs(wires=0)
+
+        res=circuit2()
+        expected = np.array([0, 1])
+        assert np.allclose(res, expected)
+
+
+    @pytest.mark.parametrize("nr_wires", [2,3])
+    def test_readout_prob_half(self,nr_wires):
+        """Tests the measurement results when readout error probability=0.5"""
+        dev = qml.device("default.mixed",wires=nr_wires,readout_prob=0.5)
+
+        @qml.qnode(dev)
+        def circuit1():
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
```

Comment:
It would also be good to use observables other than `qml.PauliZ` so that there are rotations that are being applied.